composer/modules/web/src/plugins/ballerina/views/design-view.jsx

```diff
@@ -156,6 +173,61 @@ class DesignView extends React.Component {
         return this.toolPaletteContainer;
     }
 
+    mouseUpHandle(event) {
+        if (this.state.dragging) {
+            this.diagramContainer.classList.remove('grabbing');
+
+            this.state.dragging = false;
+            this.setState(this.state);
+        }
+    }
+
+    mouseDownHandle(event) {
+        if (!this.state.dragging) {
+            this.state.dragging = true;
```

Comment:
avoid directly manipulating state like this. use 
```js
this.setState({
  dragging: true,
})
```

Reply:
oh yes.. I have missed it here. Ill fix.

Reply:
Fixed.